rts/Schedule.c

```diff
@@ -203,6 +203,60 @@ traceEventAllocated (Capability *cap, StgTSO *t, EventTypeNum event)
     traceEventThreadCounter(cap, t, event, allocated * sizeof(W_));
 }
 
+static void updateThreadCPUTimePre (StgTSO *t)
+{
+    struct timespec ts;
+    int retval;
+    retval = clock_gettime (CLOCK_THREAD_CPUTIME_ID, &ts);
+    if (retval != 0) {
+        fprintf (stderr, "clock_gettime before failed");
+    } else {
+        if (t->cur_sec < 0 || t->cur_nsec < 0) {
+            fprintf (stderr, "ON ENTRY ERROR: t->cur_sec = %ld t->cur_nsec = %ld\n", t->cur_sec, t->cur_nsec);
+        }
+        t->cur_sec -= ts.tv_sec;
```

Comment:
You do not seem to be initializing t->cur_sec and t->cur_nsec after the thread is created. it should be done in create_thread().